plugins/ScheduledReports/API.php

```diff
@@ -608,9 +609,9 @@ public function sendReport($idReport, $period = false, $date = false, $force = f
                         $report['period_param']
                     );
 
-            } catch (Exception $e) {
+            } catch (\Throwable $e) {
                 $this->enableSaveReportOnDisk = false;
-                throw $e;
+                throw new RetryableException($e->getMessage());
```

Comment:
This would actually mean sending the report is only retried when it fails to be generated. If the task fails sending the reason for e.g. some temporary mailing issue it wouldn't be retried. Not sure if that is intended.

Reply:
That's mostly intended. Failing to generate the report should always be safe to retry because we know it hasn't been sent. Mailing issues are a bit trickier as in theory there could a be an issue where the mail will send for some recipients but still return an error, in which case a retry could cause the report to be emailed three times. 

It's probably a balance between having the report email not arrive at all because of a temporary mail issue vs the risk of some partial error where multiple email reports arrive. I can add retry exception on mail send failure if we think it's a good idea?

Reply:
At least for email report there is a check that reports aren't sent twice
https://github.com/matomo-org/matomo/blob/affd3737c1c608923ec45f484ddb9afd2650aa6e/plugins/ScheduledReports/ScheduledReports.php#L327
But guess such a thing doesn't exist for sms reports, so guess it's fine not to change anything here for now. We can also adjust / refactor that when needed.